corebehrt/functional/utils.py

```diff
@@ -57,6 +77,17 @@ def get_background_length(features: dict, vocabulary) -> int:
     return background_length + 2  # +2 for [CLS] and [SEP] tokens
 
 
+def get_background_length_dd(features: dd.DataFrame, vocabulary) -> int:
+    """Get the length of the background sentence, first SEP token included."""
+    background_tokens = set([v for k, v in vocabulary.items() if k.startswith("BG_")])
+    first_pid_value = features["PID"].compute().iloc[0]
+    first_pid = features[features["PID"] == first_pid_value]
+    all_concepts_first_pid = first_pid["concept"].compute().tolist()
+    background_length = len(set(all_concepts_first_pid) & background_tokens)
+
+    return background_length + 2  # +2 for [CLS] and [SEP] tokens
```

Comment:
Fine for now to manually add `+2` for CLS and SEP, but we should think of a better way

Reply:
Well if we assume that we always have SEP tokens we could just measure the length until the first one

Reply:
Yes, perhaps finding the first SEP token and taking that as bg len?